pdns/pdns_recursor.cc

```diff
@@ -2846,7 +2846,8 @@ static void handleNewUDPQuestion(int fd, FDMultiplexer::funcparam_t& var)
           }
 
           if(g_weDistributeQueries) {
-            distributeAsyncFunction(data, boost::bind(doProcessUDPQuestion, data, fromaddr, dest, source, destination, tv, fd, proxyProtocolValues));
+            distributeAsyncFunction(data, [data = data, fromaddr, dest, source, destination, tv, fd, proxyProtocolValues]() mutable
```

Comment:
While the other changes are in places where performance does not really matter much, this one is absolutely critical. Perhaps we should benchmark that change before merging this PR? @omoerbeek ?